types/lualogging/logging.d.tl

```diff
@@ -0,0 +1,110 @@
+local enum Level
+   "DEBUG"
+   "INFO"
+   "WARN"
+   "ERROR"
+   "FATAL"
+   "OFF"
+end
+
+local type Append = function<T>(params: T, ...: any): Log
+
+local record Log
+   append: Append
+   setLevel: function (self: Log, level: Level)
+   log: function(self: Log, level: Level, ...: any)
+   getPrint: function(self: Log, level: Level): function(...: any)
+
+   debug: function(self: Log, ...: any) 
+   info: function(self: Log, ...: any) 
+   warn: function(self: Log, ...: any) 
+   error: function(self: Log, ...: any) 
+   fatal: function(self: Log, ...: any) 
+
+   DEBUG: Level
+   INFO: Level
+   WARN: Level
+   ERROR: Level
+   FATAL: Level
+   OFF: Level
+end
+
+local enum ConsoleDestination
+   "stdout"
+   "stderr"
+end
+
+local record ConsoleParameters
+   destination: ConsoleDestination
+   logPattern: string
+   logPatterns: { Level: string }
+   timestampPattern: string
+   logLevel: Level
+end
+
+local record FileParameters
+   filename: string
+   datePattern: string
+   logPattern: string
+   logPatterns: { Level: string }
+   timestampPattern: string
+   logLevel: Level
+end
+
+local record RollingFileParameters
+   filename: string
+   maxFileSize: number
+   maxBackupIndex: number
+   logPattern: string
+   logPatterns: { Level: string }
+   timestampPattern: string
+   logLevel: Level
+end
+
+local record Types
+   Level: Level
+   Append: Append
+   Log: Log
+   ConsoleDestination: ConsoleDestination
+   ConsoleParameters: ConsoleParameters
+   FileParameters: FileParameters
+   RollingFileParameters: RollingFileParameters
+end
+
+local record logging
+   _COPYRIGHT: string
+   _DESCRIPTION: string
+   _VERSION: string
+
+   DEBUG: Level
+   INFO: Level
+   WARN: Level
+   ERROR: Level
+   FATAL: Level
+   OFF: Level
+
+   new: function(append: function(self: Log, level: Level, msg: string | function), startLevel: Level): Log
```

Comment:
Here is the real reason why I had to expose types:

The `new` function wants a callback parameter with specific parameters. Either, the user can't type the parameters (because they can't reference the `Log` and `Level` types), or we would have to type them here with something more open (`any`?) which is the opposite of the idea behind using a typed language.

```lua
local log = logging.new(function(_, level, msg) print(level, msg) end, logging.DEBUG) -- argument 1: argument 1: got <unknown type>, expected Log

local log2 = logging.new(function(_: logging.__types.Log, level: logging.__types.Level, msg: string) print(level, msg) end, logging.DEBUG) -- Now it works
```